pallet/src/lib.rs

```diff
@@ -622,6 +650,10 @@ pub mod pallet {
         InsufficientBalance,
         /// Script signature failure.
         ScriptSignatureFailure,
+        /// If the script expects a list of signers, only users from that list can sign the transaction.
+        UnexpectedUserSignature,
+        /// Proper signature missing.
+        ScriptSignatureMissing,
```

Comment:
If the one case above cannot appear, and we replace that by an `unwrap()` then this additional state wouldn't be needed too.